src/auth/auth_azure_ad_oauth.py

```diff
@@ -0,0 +1,34 @@
+import logging
+
+import tornado.auth
+
+from auth.auth_abstract_oauth import AbstractOauthAuthenticator, _OauthUserInfo
+from model import model_helper
+
+LOGGER = logging.getLogger('script_server.AzureADOauthAuthenticator')
+
+
+class AzureAdOAuthAuthenticator(AbstractOauthAuthenticator):
+    def __init__(self, params_dict):
+        params_dict['group_support'] = False
+        self.auth_url = model_helper.read_obligatory(params_dict, 'auth_url', ' for OAuth')
+        self.token_url = model_helper.read_obligatory(params_dict, 'token_url', ' for OAuth')
+
+        super().__init__(
+            self.auth_url,
+            self.token_url,
+            'openid email profile',
+            params_dict,
+        )
+
+    async def fetch_user_info(self, access_token) -> _OauthUserInfo:
+        headers = {'Authorization': f'Bearer {access_token}'}
+        user_response = await self.http_client.fetch('https://graph.microsoft.com/v1.0/me', headers=headers)
```

Comment:
did you try using the same approach as in GitlabOAuthAuthenticator? I.e. extend `OAuth2Mixin` and then make a request by
```
user = await self.oauth2_request(
            'https://graph.microsoft.com/v1.0/me',
            access_token=access_token)
```

Reply:
Thanks, I will try this next week!

Reply:
Unfortunately, I wasn't able to get it to work with OAuth2Mixin. I would get unauthorized error 401. It might be that the Microsoft endpoint doesn't support the fetch methods of either embedding access token in URL or access token as POST request body.
https://learn.microsoft.com/en-us/graph/auth-v2-user?tabs=http#request

^ Their example in this link only shows using the header method.

Reply:
Thanks for trying!